### PR TITLE
Hw Wallets: Improve how missing libraries are handled

### LINKED ISSUE #1244: Ledger: Warn when btchip-python is not installed
As discussed in #925, there needs to be a warning message when `btchip-python` is not installed. Maybe there should be a general system for plugins to signal missing dependencies.

### PR DESCRIPTION
**Pull request by EchterAgo**
closes #1244

### CODE REVIEW

**Comment by EchterAgo** [comment]
@cculianu ready for review

**Comment by EchterAgo** [comment]
This PR also makes the suggested troubleshooting steps conditional based on `sys.platform`

**Comment by EchterAgo** [comment]
You can see my comments and screenshots in #1244

File: lib/base_wizard.py [hunk 1]
```diff
@@ -186,12 +190,25 @@ def choose_hw_device(self):
...
+            msgs = [
                 _('No hardware device detected.') + '\n',
-                _('To trigger a rescan, press \'Next\'.') + '\n\n',
-                _('If your device is not detected on Windows, go to "Settings", "Devices", "Connected devices", and do "Remove device". Then, plug your device again.') + ' ',
-                _('On Linux, you might have to add a new permission to your udev rules.'),
-            ])
+                _('To trigger a rescan, press \'Next\'.') + '\n\n'
+            ]
+
+            if sys.platform in ('win32', 'win64', 'windows'):
```
Comment (cculianu) [inline lib/base_wizard.py]: Yes.  Nice.  Why am I hearing about Windows when I'm on Linux or macOS? Etc? Python knows this stuff already.  That always kind of bugged me.  Thanks for this.

File: lib/base_wizard.py [hunk 2]
```diff
@@ -186,12 +190,25 @@ def choose_hw_device(self):
...
             devices += list(map(lambda x: (name, x), u))
         if not devices:
-            msg = ''.join([
+            msgs = [
                 _('No hardware device detected.') + '\n',
-                _('To trigger a rescan, press \'Next\'.') + '\n\n',
-                _('If your device is not detected on Windows, go to "Settings", "Devices", "Connected devices", and do "Remove device". Then, plug your device again.') + ' ',
-                _('On Linux, you might have to add a new permission to your udev rules.'),
-            ])
+                _('To trigger a rescan, press \'Next\'.') + '\n\n'
```
Comment (cculianu) [inline lib/base_wizard.py]: Can you do me a favor? I actually want to hear from users.  Maybe you can add something like:

"If you this problem persists (and you really do have a hardware device connected), please visit https://github.com/Electron-Cash/Electron-Cash/issues to file an issue report with the developers."

Comment (EchterAgo) [reply lib/base_wizard.py]: Will do

Comment (cculianu) [reply lib/base_wizard.py]: Thanks man you're awesome 👍

Comment (EchterAgo) [reply lib/base_wizard.py]: I added this, also made the link clickable and the text selectable.

**Comment by EchterAgo** [comment]
@cculianu I added the new message and made the pip commands selectable

**Comment by EchterAgo** [comment]
It now looks like this:

image

**Comment by EchterAgo** [comment]
Maybe we should directly link to `https://github.com/Electron-Cash/Electron-Cash/issues/new` ?

**Comment by EchterAgo** [comment]
Also, we might want to remove the "No hardware wallet support found on your system." path, it is actually redundant.

**Comment by EchterAgo** [comment]
I removed the redundant case. Now, when no HW wallet plugin is usable, the message looks like this:

image

**Comment by cculianu** [comment]
The base_wizard is potentially called/used from platforms that are not Qt.  (We may offer plugins for Androind and iOS some day).

 I'm modifying your commit to be more conservative with with what it assumes the platform can do and not embed rich text.  I know it sucks -- users can just select the link and paste it.

**Comment by cculianu** [comment]
Ok man, I sort of undid the rich text stuff and just pushed a commit.  I know, I know.. it looks worse.  But I want to respect the dictum " lib/ directory doesn't know what platform it's on".

Also .. I don't want users just clicking that link necessarily.  Unless they really are having a problem and are motivated. Otherwise we may get *too many* issue reports. ;)

Anyway thanks so much for your hard work and quick response on this.  This is a great PR.

**Comment by EchterAgo** [comment]
That's fine, I didn't actually think of EC supporting multiple platforms.

**Comment by cculianu** [comment]
It does. :)

Granted none of them use base_wizard because it's a lot of trouble to do so.  For iOS I rolled my own..  -- base_wizard was written to be theoretically re-usable anywhere .. so why break it now? :)

I admit it's a small nit that doesn't matter but I hate making things worse ..

It looks ok anyway.. and it's a corner case.  And the advantage of them having to select the link and paste it is actually less noise in the issue tracker... in my experience. :)

**Pull request merged**